Source/Core/DolphinQt2/Main.cpp

```diff
@@ -8,17 +8,31 @@
 #include <QMessageBox>
 #include <QObject>
 
+#include "Common/MsgHandler.h"
 #include "Core/Analytics.h"
 #include "Core/BootManager.h"
 #include "Core/Core.h"
 #include "DolphinQt2/Host.h"
 #include "DolphinQt2/InDevelopmentWarning.h"
 #include "DolphinQt2/MainWindow.h"
+#include "DolphinQt2/QtUtils/RunOnObject.h"
 #include "DolphinQt2/Resources.h"
 #include "DolphinQt2/Settings.h"
 #include "UICommon/CommandLineParse.h"
 #include "UICommon/UICommon.h"
 
+bool QtMsgAlertHandler(const char* caption, const char* text, bool yes_no, MsgType style)
+{
+  return RunOnObject(QApplication::instance(), [&] {
+    QMessageBox message_box(QApplication::activeWindow());
```

Comment:
Might want to call [`setIcon()`](https://doc.qt.io/qt-5.9/qmessagebox.html#icon-prop) with:
* QMessageBox::Question if style is QUESTION
* QMessageBox::Information if style is INFO
* QMessageBox::Warning if style is WARNING
* QMessageBox::Critical if style is CRITICAL

(yay, all of the styles map directly)